lib/command/update.rb

```diff
@@ -293,60 +286,15 @@ def remove_old_log
       end
     end
 
-    def update_general_lastup(through_frozen_novel: true)
-      completed = false
-      database = Database.instance
-      puts "最新話掲載日を更新しています..."
-      progressbar = ProgressBar.new(database.get_object.size - 1)
-      interval = Interval.new(@options["interval"])
-      database.each.with_index do |(id, data), i|
-        progressbar.output(i)
-        if through_frozen_novel
-          next if Narou.novel_frozen?(id)
-        end
-        setting =  Downloader.get_sitesetting_by_target(id)
-        interval.wait
-        begin
-          info = NovelInfo.load(setting)
-        rescue OpenURI::HTTPError, Errno::ECONNRESET => e
-          setting.clear
-          next
-        end
-        if info
-          dates = {
-            "general_firstup" => info["general_firstup"],
-            "novelupdated_at" => info["novelupdated_at"],
-            "general_lastup" => info["general_lastup"]
-          }
-        else
-          # 小説情報ページがない場合は目次から取得する
-          begin
-            dates = get_latest_dates(id)
-          rescue OpenURI::HTTPError, Errno::ECONNRESET => e
-            setting.clear
-            next
-          end
-        end
-        database[id].merge!(dates)
-        setting.clear
-      end
-      database.save_database
-      completed = true
-    ensure
-      progressbar.clear if progressbar
-      puts "更新が完了しました" if completed
-    end
+    def update_general_lastup(option = nil)
+      puts "最新話掲載日を確認しています..."
 
-    # オンラインの目次からgeneral_lastupを取得する
-    # ただし、toc.yaml に最新話が存在し、かつsubdateが設定されていたらそれを使う
-    def get_latest_dates(target)
-      downloader = Downloader.new(target)
-      old_toc = downloader.load_toc_file
-      latest_toc = downloader.get_latest_table_of_contents(old_toc, through_error: true)
-      {
-        "novelupdated_at" => downloader.get_novelupdated_at,
-        "general_lastup" => downloader.get_general_lastup
-      }
+      updater = GeneralLastupUpdater.new(@options)
+      updater.update_narou_novels if !option || option == "narou"
+      updater.update_other_novels if !option || option == "other"
```

Comment:
__[reek]__
* [FeatureEnvy] refers to option more than self (maybe move it to another class?) [:ref](https://github.com/troessner/reek/blob/master/docs/Feature-Envy.md)

<!-- Comment by [SideCI](https://sideci.com) -->